GitCommands/Git/GitModule.cs

```diff
@@ -2143,42 +2143,47 @@ public IReadOnlyList<PatchFile> GetInteractiveRebasePatchFiles()
 
             List<PatchFile> patchFiles = new();
 
-            if (todoCommits is not null)
+            if (todoCommits is null)
             {
-                string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
+                return patchFiles;
+            }
 
-                string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
-                var isCurrentFound = false;
-                foreach (string todoCommit in todoCommits)
-                {
-                    if (todoCommit.StartsWith(commentChar))
-                    {
-                        continue;
-                    }
+            string commentChar = EffectiveConfigFile.GetString("core.commentChar", "#");
 
-                    string[] parts = todoCommit.Split(Delimiters.Space);
+            string[][] todoCommitsInfos = todoCommits.Where(l => !l.StartsWith(commentChar))
+                .Select(l => l.Split(Delimiters.Space))
+                .Where(p => p.Length >= 3)
+                .ToArray();
 
-                    if (parts.Length < 3)
-                    {
-                        continue;
-                    }
+            string? currentCommitShortHash = File.Exists(CurrentFilePath) ? File.ReadAllText(CurrentFilePath).Trim() : null;
+            var isCurrentFound = false;
```

Comment:
Please don't use `var` for the intrinsic types.
```suggestion
            bool isCurrentFound = false;
```
Ditto below